File: BankingAppFrontend/Banking_App/src/app/account.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Account } from './account';
import { Transaction } from './Transaction';

@Injectable({
  providedIn: 'root'
})
export class AccountService {

  httpClient = inject(HttpClient);

  private baseUrl="http://localhost:8080/api/accounts"

  getAllAccounts():Observable<Account[]>{
      return this.httpClient.get<Account[]>(`${this.baseUrl}`);
  }
  createAccount(account:Account):Observable<Account>{
      return this.httpClient.post<Account>(`${this.baseUrl}`,account)

  }
 getAccountById(id:number):Observable<Account>{
  return this.httpClient.get<Account>(`${this.baseUrl}/${id}`)
}

 deposit(id:number,amount:number):Observable<Account>{
 const request={ amount };
 
  return this.httpClient.put<Account>(`${this.baseUrl}/${id}/deposit`, request);
}

Withdraw(id:number,amount:number):Observable<Account>{
  const request={ amount };
 
  return this.httpClient.put<Account>(`${this.baseUrl}/${id}/withdraw`, request);

}

delete(id:number):Observable<Account>{
 
 
  return this.httpClient.delete<Account>(`${this.baseUrl}/${id}`);
}

getStatementById(id:number):Observable<Transaction[]>{

 return this.httpClient.get<Transaction[]>(`${this.baseUrl}/${id}/statement`);

}

}
   

File: BankingApplication/src/main/java/com/BankingAPPSpringBoot/BankingApplication/service/AccountService.java
package com.BankingAPPSpringBoot.BankingApplication.service;

import java.util.List;

import com.BankingAPPSpringBoot.BankingApplication.dto.AccountDto;
import com.BankingAPPSpringBoot.BankingApplication.dto.TransactionDto;

public interface AccountService {
AccountDto createAccount(AccountDto account);
AccountDto getAccountById(Long id);
AccountDto deposit(Long id,Double amount);
AccountDto withdraw(Long id,double amount);
AccountDto transferAmount(Long id, double amount,Long tid);
List<TransactionDto> getStatementById(Long accountId);

List<AccountDto>getAllAccounts();

void deleteAccount(Long id);
}
